pkg/workload/workload.go

```diff
@@ -828,8 +828,25 @@ func PrepareWorkloadPatch(w *kueue.Workload, strict bool, clk clock.Clock) *kueu
 }
 
 // ApplyAdmissionStatusPatch applies the patch of admission related status fields of a workload with SSA.
-func ApplyAdmissionStatusPatch(ctx context.Context, c client.Client, patch *kueue.Workload) error {
-	return c.Status().Patch(ctx, patch, client.Apply, client.FieldOwner(constants.AdmissionName), client.ForceOwnership)
+func ApplyAdmissionStatusPatch(ctx context.Context, c client.Client, orig, patch *kueue.Workload) error {
+	return ApplyPatch(ctx, c, constants.AdmissionName, orig, patch)
+}
+
+// ApplyPatch applies SSA patch
+func ApplyPatch(ctx context.Context, c client.Client, fieldOwner string, orig, patch *kueue.Workload) error {
+	options := []client.SubResourcePatchOption{
+		client.ForceOwnership,
+		client.FieldOwner(fieldOwner),
+	}
+	if (len(orig.Status.NominatedClusterNames) > 0 || orig.Status.ClusterName != nil) && (patch.Status.ClusterName != nil || IsEvicted(patch)) {
+		// This is a workaround for the SSA limitation, unstructured object also failed to help here
+		return clientutil.PatchStatus(ctx, c, orig, func() (bool, error) {
+			orig.Status.ClusterName = nil
```

Comment:
This looks wrong, I think this `orig.Status.ClusterName = nil` should only happen if `IsEvicted(patch)`, no?